static/js/stream_create.js

```diff
@@ -283,35 +321,27 @@ $(function () {
     $(".subscriptions").on("submit", "#stream_creation_form", function (e) {
         e.preventDefault();
         var stream = $.trim($("#create_stream_name").val());
-        var description = $.trim($("#create_stream_description").val());
-
         var name_ok = stream_name_error.validate_for_submit(stream);
 
         if (!name_ok) {
             return;
         }
 
-        var principals = _.map(
-            $("#stream_creation_form input:checkbox[name=user]:checked"),
-            function (elem) {
-                return $(elem).val();
-            }
-        );
-
-        // You are always subscribed to streams you create.
-        principals.push(people.my_current_email());
-
-        created_stream = stream;
+        var principals = get_principals();
+        if (principals.length > 100) {
+            show_large_invites_warning(principals.length);
+        } else {
+            create_stream();
+        }
+    });
 
-        var announce = (!!page_params.notifications_stream &&
-            $('#announce-new-stream input').prop('checked'));
+    $(document).on("click", ".close-invites-warning-modal", function () {
+        $("#invites-warning-overlay").remove();
+    });
 
-        ajaxSubscribeForCreation(stream,
-            description,
-            principals,
-            $('#stream_creation_form input[name=privacy]:checked').val() === "invite-only",
-            announce
-        );
+    $(document).on("click", ".confirm-invites-warning-modal", function () {
+        create_stream();
```

Comment:
I also use create_stream() here and principals and stream are not available here that is why I have not passed them as arguments but rather calculated them in the function itself.